src/components/PendingTxsList/index.tsx

```diff
@@ -0,0 +1,191 @@
+import { ReactElement, useEffect, useState } from 'react'
+import { Icon, Text } from '@gnosis.pm/safe-react-components'
+import {
+  getTransactionQueue,
+  Transaction,
+  TransactionListItem,
+  TransactionSummary,
+} from '@gnosis.pm/safe-react-gateway-sdk'
+import { List } from '@material-ui/core'
+import ChevronRightIcon from '@material-ui/icons/ChevronRight'
+import { Link } from 'react-router-dom'
+
+import { isMultisigExecutionInfo, isTransactionSummary } from 'src/logic/safe/store/models/types/gateway.d'
+import { grey400, sm, primary200 } from 'src/theme/variables'
+import { CustomIconText } from 'src/components/CustomIconText'
+import { getTxTo } from 'src/routes/safe/components/Transactions/TxList/utils'
+
+import styled from 'styled-components'
+import Spacer from 'src/components/Spacer'
+import { useSelector } from 'react-redux'
+import { GATEWAY_URL } from 'src/utils/constants'
+import { currentChainId } from 'src/logic/config/store/selectors'
+import { lastViewedSafe } from 'src/logic/currentSession/store/selectors'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import useOwnerSafes from 'src/logic/safe/hooks/useOwnerSafes'
+import { useTransactionType } from 'src/routes/safe/components/Transactions/TxList/hooks/useTransactionType'
+import { useKnownAddress } from 'src/routes/safe/components/Transactions/TxList/hooks/useKnownAddress'
+import { generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+import { getChainById } from 'src/config'
+import { useAssetInfo } from 'src/routes/safe/components/Transactions/TxList/hooks/useAssetInfo'
+import { TxInfo } from 'src/routes/safe/components/Transactions/TxList/TxCollapsed'
+
+const MAX_TXS_DISPLAY = 5
+
+const PendingTxsList = (): ReactElement => {
+  const chainId = useSelector(currentChainId)
+  const defaultSafe = useSelector(lastViewedSafe) || undefined
+
+  const { isLoading: isLoadingOwnerSafes, ownerSafes } = useOwnerSafes()
+  const [loadingOwnerSafes, setLoadingOwnerSafes] = useState<boolean>()
+
+  // Selected Safe should come from the SideBar task when it's done
+  const [selectedSafe, setSelectedSafe] = useState<string>()
+
+  const [isFetchingPendingTxs, setIsFetchingPendingTxs] = useState<boolean>()
+  const [queuedTransactions, setQueuedTransactions] = useState<TransactionSummary[]>([])
+
+  useEffect(() => {
+    if (isLoadingOwnerSafes === undefined) return
+
+    if (ownerSafes[chainId]?.length) {
+      setSelectedSafe(ownerSafes[chainId][0])
+    } else {
+      setSelectedSafe(defaultSafe)
+    }
+    setLoadingOwnerSafes(false)
+  }, [chainId, defaultSafe, isLoadingOwnerSafes, ownerSafes, selectedSafe])
+
+  // Fetch txs awaiting confirmations
+  useEffect(() => {
+    if (selectedSafe === undefined) return
+
+    let isCurrent = true
+    const fetchQueuedTxs = async () => {
+      const { results } = await getTransactionQueue(GATEWAY_URL, chainId, checksumAddress(selectedSafe))
```

Comment:
Coming back to this after the daily: the MVP was supposed to be singular Safe-based. Can we not just use the local state we have? This is an unnecessary fetch imo. We could end up with modules individually requesting the same data.